### DISCUSSION TITLE
Raycasting against shapes, seems to always be against the bounding sphere?

### DISCUSSION THREAD

**Discussion by DennisSmolek** [jrouwe/JoltPhysics.js#144] [Q&A] [answered]
I'm working on replicating the [raycasting demo from three-mesh-bvh](https://github.com/gkjohnson/three-mesh-bvh) and noticed something strange...
Original:
![image](https://github.com/jrouwe/JoltPhysics.js/assets/1397052/22d814ad-879d-46e4-835c-adf415147778)

Mine:
![Screenshot 2024-04-05 003639](https://github.com/jrouwe/JoltPhysics.js/assets/1397052/cbed35cc-1aaf-405c-914d-3f024d61716a)

I originally thought I was just getting values wrong or there was some sort of offset, however I debugged with the bounding sphere:
![Screenshot 2024-04-05 003711](https://github.com/jrouwe/JoltPhysics.js/assets/1397052/42124d40-4882-4c86-9e7b-bc4a2c1f5878)

And it's totally correct!

For some reason no matter the shape ( box, convex, capsule) the raycasts always get marked when impacting the bounding sphere of the shape. 

I'm using the `Jolt.CastRayClosestHitCollisionCollector;`

My theories are I'm missing some setting or flag that tracks the bounding sphere in the ray
or I'm doing something else wrong...

**Comment by jrouwe** [comment]
I have no idea how this is possible, the mesh shape doesn't use bounding spheres at all (it uses an aabb). Maybe you have a repro or some code I can take a look at?

**Reply by DennisSmolek** [reply]
I'll try to reproduce using vanilla JS tonight

**Reply by DennisSmolek** [reply, marked as answer]
I think I figured this out...
I think this comes down to a fundamental difference in how threeJS and Jolt handle raycasts.

I'm not positive, but I believe that because Jolt's ray's work on ranges from the origin to the point in the direction, that the ray's were ending prematurely. So as the ray moves, it shoot's to the center and reaches the end before touching the shape. 

In a threeJS ray cast you normalize the direction `raycaster.set(origin, direction.normalize)` but the ray goes forever. So the original code included the normaizing of the direction ray. I removed that and things seem to be working correctly.

**Reply by LeXXik** [reply]
You can reset the length of the ray:

https://github.com/jrouwe/JoltPhysics.js/blob/a85e6496b9b067258cb3f3a1bce3618a549cd87b/Examples/ray_cast.html#L74-L83

**Reply by DennisSmolek** [reply]
What I mean was Jolt expects direction to be the end point of the ray, so `raycast([0,0,0], [10, 10, 10])` would stop results from after 10, 10, 10. 
Where for threejs `[10, 10, 10]` and `[1,1,1]` are the same. once normalized. 

The original code expected the rays to go forever so didn't factor that into the direction calculation.
